probatus/missing/imputation.py

```diff
@@ -0,0 +1,318 @@
+        # columns
+        X_num = X.drop(columns = categorical_columns,inplace=False)
+        numeric_columns = X_num.columns.to_list()
+        
+        for strategy in self.strategies:
+
+            numeric_transformer = Pipeline(steps=[
+                    ('imputer', self.strategies[strategy])])
+
+            categorical_transformer = Pipeline(steps=[
+                    ('imp_cat',SimpleImputer(strategy='constant',fill_value='missing',add_indicator=True)),
+                    ('ohe_cat',OneHotEncoder(handle_unknown='ignore')),
+                ])
+
+            preprocessor = ColumnTransformer(
+                    transformers=[
+                        ('num', numeric_transformer, numeric_columns),
+                        ('cat', categorical_transformer, categorical_columns)],
+                        remainder='passthrough')
+
+            clf = Pipeline(steps=[('preprocessor', preprocessor),
+                                    ('classifier', self.clf)])
+                
+            imputation_results = cross_val_score(
+                clf,
+                X,
+                y,
+                scoring=self.scorer.scorer,
+                cv=self.cv,
+                n_jobs = self.n_jobs)
+
+            temp_results = {
+                    'strategy' : strategy,
+                    'score': np.round(np.mean(imputation_results),3),
+                    'std':np.round(np.std(imputation_results),3),
+                }
+            results.append(temp_results)
+        #If model supports missing values by default, then calculate the scores 
+        #on raw data without any imputation. 
+        if self.model_na_support :
+            categorical_transformer = Pipeline(steps=[
+                    ('ohe_cat',OneHotEncoder(handle_unknown='ignore')),
+                ])
+
+            preprocessor = ColumnTransformer(
+                    transformers=[
+                        ('cat', categorical_transformer, categorical_columns)],
+                        remainder='passthrough')
+
+            self.clf = Pipeline(steps=[('preprocessor', preprocessor),
+                                    ('classifier', self.clf)])
+
+            imputation_results = cross_val_score(
+                self.clf,
+                X,
+                y,
+                scoring=self.scorer.scorer,
+                cv=self.cv,
+                n_jobs = self.n_jobs
+                )
+
+            temp_results = {
+                    'strategy' : 'Model Imputation',
+                    'score': np.round(np.mean(imputation_results),3),
+                    'std':np.round(np.std(imputation_results),3),
+                }
+            results.append(temp_results)
+
+
+        self.report = pd.DataFrame(results)
+        self.report.sort_values(by='score',inplace=True)
+        self.fitted = True
+        return self
+        
+        
+    def compute(self,return_scores=True):
```

Comment:
You don't need parameter return_scores. It is used if you also want to return AUC next to the report.

In this case compute should always return the report, to be consistent with the API

Reply:
Done.